src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEF_CAPACITY = 10;
+    private Object[] array;
+    private int size;
 
+    public MateStack() {
+        array = new Object[DEF_CAPACITY];
+        size = 0;
+    }
+
+    public void push(T value) {
+        checkCapacity();
+        array[size++] = value;
     }
 
     public T peek() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+        return (T) array[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
```

Comment:
better extract to a separate method not only condition checking, but exception throwing too